### ISSUE TITLE
Mod template

### ISSUE THREAD

**Issue by ki-foobar** [ElonaFoobar/ElonaFoobar#1284]
# Summary

We need a way to create mods with ease. Mod template is a template of new mod. When you select "Create mod" menu in mod menu, foobar creates a new folder for the mod and copies the contents from the template to the new mod.

# Details

Template mod is placed in other folder or named in special convention. For example,

- Mods under `mod/template` are template mods.
- Mods having "template_" prefix are template mods.

They are listed when you select "Create mod" menu.

Template mods can have script to do something else.

```lua
-- template.lua, called after copying the contents.

copy_additional_files()
remove_unneccessary_files()
...
```

**Comment by Vorlent** [comment]
I don't think you should include your own name in the template: https://github.com/ElonaFoobar/ElonaFoobar/blob/develop/runtime/mod/template_blank/mod.hcl

If someone forgets to change it you will show up as the author of the mod.

**Comment by Vorlent** [comment]
Elona Foobar doesn't load mod folders that contain upper case characters but it is possible to enter them in the template creation dialog and therefore create a mod that will never be loaded.

**Comment by ki-foobar** [comment]
> I don't think you should include your own name in the template: https://github.com/ElonaFoobar/ElonaFoobar/blob/develop/runtime/mod/template_blank/mod.hcl
> 
> If someone forgets to change it you will show up as the author of the mod.

Makes sense. I'll fix it.

> Elona Foobar doesn't load mod folders that contain upper case characters but it is possible to enter them in the template creation dialog and therefore create a mod that will never be loaded.

Oh, thanks. I'll fix the validation function.
